### PR TITLE
feat(sdk): submission requirements doc and usage demonstration

### PR DESCRIPTION
**Pull request by vkubiv**

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: cmd/wallet-sdk-gomobile/docs/usage.md [hunk 1]
```diff
@@ -556,7 +563,15 @@ let interaction = Openid4vpInteraction("YourAuthRequestURIHere", config: clientC
 let query = interaction.getQuery()
 let inquirer = CredentialNewInquirer(docLoader)
 let issuedCredentials = ApiVerifiableCredentialsArray() // Would need some actual credentials for this to actually work
-let verifiablePres = inquirer.Query(query, CredentialNewCredentialsOpt(issuedCredentials))
+
+// Use this code to display the list of VCs to select which of them to send.
+let matchedRequirements = inquirer.getSubmissionRequirements(query, credentials) 
+let matchedRequirement = matchedRequirements.atIndex(0) // Usually we will have one requirement
```
Comment (sudeshrshetty) [inline cmd/wallet-sdk-gomobile/docs/usage.md:569]: how client applications will find out about number of requirements?

File: cmd/wallet-sdk-gomobile/docs/usage.md [hunk 2]
```diff
@@ -556,7 +563,15 @@ let interaction = Openid4vpInteraction("YourAuthRequestURIHere", config: clientC
...
+let requirementDesc = matchedRequirement.descriptorAtIndex(0) // Usually requirement will contain one descriptor
+let selectedVCs = ApiVerifiableCredentialsArray()
+selectedVCs.add(requirementDesc.matchedVCs.atIndex(0)) // Users should select one VC for each descriptor from the matched list and confirm that they want to share it
+
+let verifiablePres = inquirer.Query(query, CredentialNewCredentialsOpt(selectedVCs))
```
Comment (sudeshrshetty) [inline cmd/wallet-sdk-gomobile/docs/usage.md:574]: Option name stutters, change it to something like `WithSelections`,

Also inquirer.Query is confusing, rename it to something like `inquirer.Present or inquirer.prove`

File: cmd/wallet-sdk-gomobile/docs/usage.md [hunk 3]
```diff
@@ -556,7 +563,15 @@ let interaction = Openid4vpInteraction("YourAuthRequestURIHere", config: clientC
 let query = interaction.getQuery()
 let inquirer = CredentialNewInquirer(docLoader)
 let issuedCredentials = ApiVerifiableCredentialsArray() // Would need some actual credentials for this to actually work
-let verifiablePres = inquirer.Query(query, CredentialNewCredentialsOpt(issuedCredentials))
+
+// Use this code to display the list of VCs to select which of them to send.
+let matchedRequirements = inquirer.getSubmissionRequirements(query, credentials)
```
Comment (sudeshrshetty) [inline cmd/wallet-sdk-gomobile/docs/usage.md:568]: so actual presentation exchange match is performed here??

Comment (vkubiv) [reply cmd/wallet-sdk-gomobile/docs/usage.md:568]: Yes it matches the given credentials against the presentation

File: cmd/wallet-sdk-gomobile/docs/usage.md (hunk 2 above)
Comment (vkubiv) [reply cmd/wallet-sdk-gomobile/docs/usage.md:574]: CredentialNewCredentialsOpt is a typo on the documentation, it is actually CredentialCredentialsOpt. We don't have much control over how go mobile generates names. For swift it takes package name 'credential' concatenate with the name of CredentialsOpt struct. for kotlin it just leaves CredentialsOpt as is. Also, CredentialsOpt is a struct to support multiple input options for credentials, as an array or as the credential reader. it not possible to implement 'With' pattern with gomobile

File: cmd/wallet-sdk-gomobile/docs/usage.md (hunk 3 above)
Comment (vkubiv) [reply cmd/wallet-sdk-gomobile/docs/usage.md:568]: verifiable presentation is generating few lines below

**Review by sudeshrshetty — APPROVED** [review]

**Pull request merged**